### PR TITLE
Upgrade clang-format to 3.8

### PR DESCRIPTION
**Pull request by ShikharJ**

### CODE REVIEW

File: symengine/complex_mpc.cpp [hunk 1]
```diff
@@ -11,12 +11,10 @@
...
 
-ComplexMPC::ComplexMPC(mpc_class i) : i{std::move(i)}
-{
-    SYMENGINE_ASSIGN_TYPEID()
-}
+ComplexMPC::ComplexMPC(mpc_class i)
+    : i{std::move(i)} {SYMENGINE_ASSIGN_TYPEID()}
 
-hash_t ComplexMPC::__hash__() const
+      hash_t ComplexMPC::__hash__() const
```
Comment (isuruf) [inline symengine/complex_mpc.cpp]: Why this change?

File: symengine/functions.cpp [hunk 2]
```diff
@@ -1701,13 +1701,11 @@ RCP<const Basic> lambertw(const RCP<const Basic> &arg)
...
-{
-    SYMENGINE_ASSIGN_TYPEID()
-    SYMENGINE_ASSERT(is_canonical(get_vec()))
-}
+    : MultiArgFunction({arg}), name_{name} {SYMENGINE_ASSIGN_TYPEID()
+                                                SYMENGINE_ASSERT(
+                                                    is_canonical(get_vec()))}
 
-FunctionSymbol::FunctionSymbol(std::string name, const vec_basic &arg)
+      FunctionSymbol::FunctionSymbol(std::string name, const vec_basic &arg)
```
Comment (isuruf) [inline symengine/functions.cpp]: Why?

File: symengine/functions.cpp [hunk 3]
```diff
@@ -1764,20 +1762,15 @@ RCP<const Basic> function_symbol(std::string name, const RCP<const Basic> &arg)
...
-    SYMENGINE_ASSIGN_TYPEID()
-}
+    : FunctionSymbol(name, arg){SYMENGINE_ASSIGN_TYPEID()}
 
-FunctionWrapper::FunctionWrapper(std::string name, const vec_basic &vec)
-    : FunctionSymbol(name, vec)
-{
-    SYMENGINE_ASSIGN_TYPEID()
-}
+      FunctionWrapper::FunctionWrapper(std::string name, const vec_basic &vec)
```
Comment (isuruf) [inline symengine/functions.cpp]: Why?

File: symengine/integer.h [hunk 4]
```diff
@@ -25,11 +25,9 @@ class Integer : public Number
     IMPLEMENT_TYPEID(INTEGER)
     //! Constructor of Integer using `integer_class`
     // explicit Integer(integer_class i);
-    Integer(const integer_class &_i) : i(_i)
-    {
-        SYMENGINE_ASSIGN_TYPEID()
-    }
-    Integer(integer_class &&_i) : i(std::move(_i))
+    Integer(const integer_class &_i)
+        : i(_i){SYMENGINE_ASSIGN_TYPEID()} Integer(integer_class && _i)
```
Comment (isuruf) [inline symengine/integer.h]: Why?

**Comment by ShikharJ** [comment]
@isuruf All the changes were made on using `./bin/test_format_local.sh` using `clang-format-4.0.1-6`. Any idea how can I change the ones you mentioned and fix the [installation error](https://travis-ci.org/symengine/symengine/jobs/291138736) in `Travis`?

**Comment by ShikharJ** [comment]
@isuruf Can you help me regarding this? It's inconvenient to correct the formatting manually, each time I push a new PR.

File: .clang-format [hunk 5]
```diff
@@ -53,6 +53,7 @@ PenaltyBreakString: 2123
 PenaltyExcessCharacter: 1000000
 PenaltyReturnTypeOnItsOwnLine: 60
 PointerAlignment: Right
+SortIncludes: false
```
Comment (isuruf) [inline .clang-format]: You need to change the first line of this file to 3.8 as well

File: symengine/polys/msymenginepoly.h [hunk 6]
```diff
@@ -465,8 +465,7 @@ class MIntPoly : public MSymEnginePoly<MIntDict, MIntPoly>
         SYMENGINE_ASSIGN_TYPEID()
     }
 
-    IMPLEMENT_TYPEID(MINTPOLY)
-    hash_t __hash__() const;
+    IMPLEMENT_TYPEID(MINTPOLY) hash_t __hash__() const;
```
Comment (isuruf) [inline symengine/polys/msymenginepoly.h]: This is unnecessary. Can you figure out a workaround?

File: symengine/polys/msymenginepoly.h [hunk 7]
```diff
@@ -482,8 +483,7 @@ class MExprPoly : public MSymEnginePoly<MExprDict, MExprPoly>
         SYMENGINE_ASSIGN_TYPEID()
     }
 
-    IMPLEMENT_TYPEID(MEXPRPOLY)
-    hash_t __hash__() const;
+    IMPLEMENT_TYPEID(MEXPRPOLY) hash_t __hash__() const;
```
Comment (isuruf) [inline symengine/polys/msymenginepoly.h]: Same here

Comment (ShikharJ) [reply symengine/polys/msymenginepoly.h]: I'll update this in a recent PR.
